synapse/cortex.py

```diff
@@ -2955,11 +2966,15 @@ def loadStormPkg(self, pkgdef):
             self._initEasyPerm(gdef)
             self.pkggraphs[gdef['iden']] = gdef
 
-    def _runStormPkgOnload(self, pkgdef):
+    async def _runStormPkgOnload(self, pkgdef):
         name = pkgdef.get('name')
         onload = pkgdef.get('onload')
 
         if onload is not None and self.isactive:
+            if self.safemode:
```

Comment:
Is this event here for testing?

Reply:
We could move the fire into the `_onload` coro and we'd be able to leave this as a sync function?

```
async def _onload():
    if self.safemode:
        await self.fire('core:pkg:onload:skipped', pkg=name, reason='safemode')
                return
        ...
```

Reply:
yeah, just for testing. There was a reason I didn't put it in _onload but I don't remember what that was now. I can move it.

Reply:
One of the things that comes to mind is that addpkg handler is sync code up until the end, which allows the full replacement of a package without a storm runtime getting two different versions of the package mid-import, or a package that has been removed and not yet added.